fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -242,7 +245,14 @@ private static Boolean compareEquals(Object value, Object comparand) {
         if (value == null || comparand == null) {
             return null;
         } else {
-            return toClassWithRealEquals(value).equals(toClassWithRealEquals(comparand));
+            if (value instanceof Message) {
```

Comment:
Having two protobuf messages be equal that happen to be shaped the same but come from different descriptors sounds suspicious in general. At the least, it will probably hide bugs in metadata maintenance. Is there no way to restrict it to whatever special case this is arising from?